src/DeepLinkModule/javasource/deeplink/actions/ExecuteDeeplink.java

```diff
@@ -113,8 +113,9 @@ public java.lang.Boolean executeAction() throws Exception
 
 			//remove the pendinglink, unless it should be reused during this session..
 			if (link.getUseAsHome()) { //do not remove if used as home.
-				this.pendinglink.setSessionId(this.getContext().getSession().getId().toString());
-				Core.commit(this.getContext(), this.pendinglink.getMendixObject());
+				IContext sudoContext = getContext().createSudoClone();

```

Comment:
We are already creating a sudo context on line 125. Might as well pull that up and reuse it here.